monai/handlers/clearml_handlers.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from monai.utils import optional_import
+
+from .tensorboard_handlers import TensorBoardImageHandler, TensorBoardStatsHandler
+
+if TYPE_CHECKING:
+    from clearml import Task
+else:
+    Task = optional_import("clearml.Task", name="Task")
+
+
+class ClearMLHandler:
+    """
+    Base class for the handlers to log everything to ClearML WebUI.
+    For more details of ClearML usage, please refer to:
+    https://clear.ml/docs/latest/docs/references/sdk/task
+
+    """
+
+    def __init__(
+        self,
+        project_name: str,
+        task_name: str,
+        output_uri: str,
+        tags,
+        reuse_last_task_id: bool,
+        continue_last_task: bool,
+        auto_connect_frameworks,
+    ):
+        """
+        Args:
+            project_name:               ClearML project name.
+            task_name:                  ClearML task name.
+            output_uri:                 The default location for output models and other artifacts.
+            tags:                       A list of tags (str) to the created Task.
+            reuse_last_task_id:         Force a new Task (experiment) with a previously used Task ID.
+            continue_last_task:         Continue the execution of a previously executed Task (experiment).
+            auto_connect_frameworks:    Automatically connect frameworks.
+        """
+
+        if Task.current_task():
+            self.clearml_task = Task.current_task()
+        else:
+            self.clearml_task = Task.init(
+                project_name=project_name,
+                task_name=task_name,
+                output_uri=output_uri,
+                tags=tags,
+                reuse_last_task_id=reuse_last_task_id,
+                continue_last_task=continue_last_task,
+                auto_connect_frameworks=auto_connect_frameworks,
+            )
+
+
+class ClearMLStatsHandler(ClearMLHandler, TensorBoardStatsHandler):
+    """
+
+    Class to write tensorboard stats by inheriting TensorBoardStatsHandler class.
+    Everything from Tensorboard is logged automatically to ClearML WebUI.
+
+    """
+
+    def __init__(
+        self,
+        project_name: str = None,
+        task_name: str = None,
```

Comment:
Hi @skinan ,
      I would suggest that we'd better add some default value like `mona_project` and `task_xxyyzz` by default, so that the record make sense to the users who don't specifiy these parameters. Same comments on `ClearMLImageHandler` class.

Thanks,
Bin

Reply:
Thank you for the reviews. I accept this review suggestion.